### PR TITLE
Prevent `__len__()` and `__getitem__()` from being overridden

### LINKED ISSUE #10: NotImplementedError
ic
hello, there is a problem occurred when i run the command 
" python train.py \
  --dataset=smnist \
  --batch_size=16 \
  --max_epochs=100 \
  --lr=1e-2 \
  --n_blocks=6 \
  --d_model=128 \
  --norm_type=layer"
The class SMnistDataset is based on the class SequenceDataset . The __len__in class SequenceDataset is not implemented, but the function (train_val_split) that divided the training set and the validation set needs len.

### PR DESCRIPTION
**Pull request by TariqAHassan**
Fixes #10

### CODE REVIEW

**Comment by TariqAHassan** [comment]
Tested. Appears to resolve the issue. Merging.
